libs/language-server/src/test/validation/checks/pipe-definition.spec.ts

```diff
@@ -0,0 +1,216 @@
+
+    it('should have no error even if pipe references block of non existing type', async () => {
+      const text = `
+      pipeline Test {
+        block UnknownOutput oftype UnknownOutputType {
+        }
+
+        block UnknownInput oftype UnknownInputType {
+        }
+
+        pipe {
+          from: UnknownOutput;
+          to: UnknownInput;
+        }
+      }
+      `;
+
+      const parseResult = await parse(text);
+
+      const pipe: PipeDefinition = extractPipe(parseResult);
+
+      validatePipeDefinition(
+        pipe,
+        new ValidationContext(validationAcceptorMock),
+      );
+
+      expect(validationAcceptorMock).toHaveBeenCalledTimes(0);
+    });
+
+    it('error on unsupported pipe between Blocktypes', async () => {
+      const text = `
+      pipeline Test {
+        block CarsExtractor oftype HttpExtractor {
+          url: "https://gist.githubusercontent.com/noamross/e5d3e859aa0c794be10b/raw/b999fb4425b54c63cab088c0ce2c0d6ce961a563/cars.csv";
+        }
+
+        block CarsLoader oftype SQLiteLoader {
+          table: "Cars";
+          file: "./cars.sqlite";
+        }
+
+        pipe {
+          from: CarsExtractor;
+          to: CarsLoader;
+        }
+      }
+      `;
+
+      const parseResult = await parse(text);
+
+      const pipe: PipeDefinition = extractPipe(parseResult);
+
+      validatePipeDefinition(
+        pipe,
+        new ValidationContext(validationAcceptorMock),
+      );
+
+      expect(validationAcceptorMock).toHaveBeenCalledTimes(2);
+      expect(validationAcceptorMock).toHaveBeenNthCalledWith(
+        2,
+        'error',
+        `The output type "File" of HttpExtractor is incompatible with the input type "Table" of SQLiteLoader`,
+        expect.any(Object),
+      );
+    });
+  });
+
+  describe('chained pipe', () => {
+    // This test should succeed, because the error is thrown by langium during linking, not during validation!
+    it('should have no error even if pipe references non existing block', async () => {
+      const text = `
+      pipeline Test {
+        CarsExtractor -> CarsLoader;
+      }
+      `;
+
+      const parseResult = await parse(text);
+
+      const pipe: PipeDefinition = extractPipe(parseResult);
+
+      validatePipeDefinition(
+        pipe,
+        new ValidationContext(validationAcceptorMock),
+      );
+
+      expect(validationAcceptorMock).toHaveBeenCalledTimes(0);
+    });
+
+    it('should have no error even if pipe references block of non existing type', async () => {
+      const text = `
+      pipeline Test {
+        block UnknownOutput oftype UnknownOutputType {
+        }
+
+        block UnknownInput oftype UnknownInputType {
+        }
+        
+        UnknownOutput -> UnknownInput;
+      }
+      `;
+
+      const parseResult = await parse(text);
+
+      const pipe: PipeDefinition = extractPipe(parseResult);
+
+      validatePipeDefinition(
+        pipe,
+        new ValidationContext(validationAcceptorMock),
+      );
```

Comment:
Here we have the same logic in every method (except setting up the model and doing the assertions). Speaking in the classic 3A pattern (arrange, act, assert), we could pull out the whole "act" as a method making the tests more compact and easier to read if we name the pulled out method well.